source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -213,12 +213,13 @@ void Filter::onComplete(Filters::Common::ExtAuthz::ResponsePtr&& response) {
 
   switch (response->status) {
   case CheckStatus::OK: {
-    // Any changes to request headers can affect how the request is going to be
+    // Any changes to request headers or query parameters can affect how the request is going to be
```

Comment:
Thanks for this. I titally forgot that we have a route matcher to match query params. https://github.com/envoyproxy/envoy/blob/7d3057f8c039910f06c583684642c6e0a4e35b37/api/envoy/config/route/v3/route_components.proto#L1995.